bookwyrm/views/list.py

```diff
@@ -200,6 +210,63 @@ def post(self, request, list_id):
         return redirect(book_list.local_path)
 
 
+class EmbedList(View):
+    """embeded book list page"""
+
+    def get(self, request, list_id, list_key):
+        """display a book list"""
+        book_list = get_object_or_404(models.List, id=list_id)
```

Comment:
I think it makes sense to check `book_list.raise_visible_to_user(request.user)` to ensure that the viewer is authorized to see the list

Reply:
I chose not to do it, based on a use case that might happen:

I'm a BookWyrm user who has a password-protected private blog. On Bookwyrm I make a private list, that I want to embed on my blog, so that the readers of my blog—non Bookwyrm users—will be able to see it. Checking `book_list.raise_visible_to_user(request.user)` for the display of the embed will guarantee that my friends won't be able to see the contents of this private list.

On the other hand, the use of an UUID key that's unique to that list will keep brute-force attempts at displaying private lists (by guessing the list id). That key is only displayed to the list creator if the list is private. There could be further limits to the visibility of the embed code for non-public lists. An alternative (that would complicate things) would be to generate a new key for each share (I think that's what Nextcloud or Google Documents do)—so that the key can be easily revoked.

On the other hand, if that use case is judged too risky for privacy, we'll have to restrict the embedding to Public lists only.

Reply:
Gotcha! I think what you've done makes sense, carry on :)